addons/website_slides/models/slide_slide.py

```diff
@@ -491,6 +491,13 @@ def write(self, values):
             values['is_preview'] = True
             values['is_published'] = True
 
+        # if the slide type is changed, remove the existing url or html_content
+        if values.get('slide_type'):
+            if self.url:
+                self.url = False
+            elif self.html_content:
+                self.html_content = False
+
```

Comment:
Hello @awa-odoo 😊

Indeed first approach was to use compute but somehow it works partially.
When we change 'slide_type' from 'article' to 'video' it gives the same constraint error.
This occurs because when using compute, the field is not written until the save button is clicked. In the case of the video, the URL field becomes mandatory, so when saving, it first writes the URL, resulting in a constraint error.

Or am I missing something ?


Thanks !!

Reply:
Hello @mrsr-odoo 

I would have suggested to try "precompute=True" but it's not available in version 15 I believe.

Then I'm not entirely sure how to proceed, I've never had this kind of issues.
But writing on the fields inside the write is something I would really like to avoid.

Can you ask guidance to the python framework team (on Discord)?